tests/integration/docker_utils/test_docker.py

```diff
@@ -710,6 +711,40 @@ def test_list_containers_filter(self, docker_client: ContainerClient, create_con
         assert 1 == len(container_list)
         assert c1.container_name == container_list[0]["name"]
 
+    def test_list_containers_with_podman_image_ref_format(
+        self, docker_client: ContainerClient, create_container, cleanups, monkeypatch
+    ):
+        # create custom image tag
+        image_name = f"alpine:tag-{short_uid()}"
+        docker_client.tag_image("alpine", image_name)
+        cleanups.append(lambda: docker_client.remove_image(image_name))
+
+        def container_init(self, attrs=None, *args, **kwargs):
+            # Simulate podman API response, Docker returns "sha:..." for Image, podman returns "<image-name>:<tag>".
+            #  See https://github.com/containers/podman/issues/8329
+            attrs["Image"] = image_name
+            container_init_orig(self, attrs=attrs, *args, **kwargs)
+
+        # apply patch to simulate podman behavior
+        container_init_orig = Container.__init__
```

Comment:
This is probably my issue, but I find it really confusing to have a variable used (`container_init_orig`) _before_ it has been defined. I know Python's scoping rules allow this, but still it's weird. Not that you need to change it, just raising for discussion.

... Or you could move the function definition to in between lines 728 and 729?

Reply:
Good point, I've gotten so used to Python's scoping that I don't notice the difference anymore (although you already pointed it out to me several times!). 😄 But makes sense, updated it! 👍